docs/REST API/rest-api.md

```diff
@@ -9014,14 +9021,89 @@ Get statuses and host counts for a batch-run script.
   "ran": 12345,
   "pending": 234,
   "errored": 18,
+  "incompatible": 3,
   "canceled": 2,
   "targeted": 12599,
   "script_id": 555,
   "script_name": "my-script.sh",
-  "team_id": 123
+  "team_id": 123,
+  "not_before": "2025-07-01T15:00:00Z",
+  "completed_at": "2025-07-06T15:00:00Z",
+  "status": "completed",
+  "cancelled": false
+}
+```
+
+
+### List batch scripts
+
+> TODO: API to list batch script executions. Paginated, with status filter (started, scheduled, completed).
+
+TODO: Description text here.
+
+`GET /api/v1/fleet/scripts/batch`
+
+#### Parameters
+
+| Name            | Type    | In   | Description                                                                                    |
+| ----            | ------- | ---- | --------------------------------------------                                                   |
+| team_id         | integer | query | _Available in Fleet Premium_. Filters to batch script runs for the specified team. |
+| status          | string  | query | Filters to batch script runs with this status. Either `"started"`, `"scheduled"`, or "`completed`". |
+| page            | integer | query | Page number of the results to fetch. |
+| per_page        | integer | query | Results per page. |
+
+
+#### Example
+
+`GET /api/v1/fleet/scripts/batch`
+
+##### Request body
+
+```json
+{
+  "team_id": 123,
+  "status": "completed"
+}
+```
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "batch_executions": [
+    {
+      "ran": 12345,
+      "pending": 234,
+      "errored": 18,
+      "incompatible": 3,
+      "canceled": 2,
+      "targeted": 12599,
+      "script_id": 555,
+      "script_name": "my-script.sh",
+      "team_id": 123,
+      "not_before": "2025-07-01T15:00:00Z",
+      "completed_at": "2025-07-06T15:00:00Z",
+      "status": "completed",
+      "cancelled": false
+    }
+  ],
+  "meta": {
+    "has_next_results": false,
+    "has_previous_results": false
+  }
 }
 ```
 
+### List batch script results
+
+> TODO: API to list hosts & their results. Paginated, with status filter (ran, pending, errored, incompatible, canceled).
+
+### Cancel batch script
+
+> TODO: API to cancel all pending script executions from a batch run script.
```

Comment:
`DELETE /scripts/batch/{batch_execution_id:[a-zA-Z0-9-]+}` ?

Reply:
That's a good point. I don't think we'd ever allow un-cancelling, and the endpoint will be idempotent, but it's still not technically deleting anything. `POST /scripts/batch/{batch_execution_id:[a-zA-Z0-9-]+}/cancel` works.